### ISSUE TITLE
IPFS swam peers no response

### ISSUE THREAD

**Issue by kevinmartinmauro** [ipfs/go-ipfs#2540]
-- IPFS daemon is running another window. Firewall has unblocked ipfs.exe
-- "ipfs swarm peers" returns no response
-- set-up was followed as described in http://ipfs.io/docs/getting-started/
-- Files are able to be hosted locally, it's just the connection part

This is using Windows 64-bit binary. About to try it with 32-bit. 

Contrarily to the issue below, "ipfs swarm peers" returns no response, not an error response. 

https://github.com/ipfs/go-ipfs/issues/1758

**Comment by mitzip** [comment]
Same here after upgrade to 0.4

**Comment by kevinmartinmauro** [comment]
I switched to 32-bit binary, and also waited about 10minutes starting the Daemon, and I ended up connecting. Not sure which of these was the trick

**Comment by sterpe** [comment]
I have same 
```
$ ipfs swarm peers
$
```
Building from source on OS X El Capitan

**Comment by mitzip** [comment]
$ lsof -i -n -P | grep ipfs

The above command shows lots of established connections for ipfs, but 

$ ipfs swarm peers

Shows nothing

I got mine as a pre-built binary.

**Comment by sterpe** [comment]
I am seeing some peers now with (0.4.1-dev)

**Comment by kevinmartinmauro** [comment]
^  Yea I think it just takes a while to connect. I will close this issue later this weekend after being sure

**Comment by sterpe** [comment]
If I kill the daemon then restart it -- Same thing...30/40 minutes or longer before swarm peers reports anything.

**Comment by interfect** [comment]
It shouldn't take this long to connect, though. 0.3 didn't. I would class this as a regression, though whether it's really a code thing or caused by the bootstrappers (or just what happens when you don't have many compatible peers) remains to be determined.

**Comment by interfect** [comment]
OK, I think what's happening is that all the bootstrap peers are still running v0.3 on the default ports, and v0.4 on the alternate ports. I did this and managed to get connected immediately:

First start the daemon:
```
ipfs daemon
```

In another terminal, remove all the bootstrap peers:
```
ipfs bootstrap rm /ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ
ipfs bootstrap rm /ip4/104.236.151.122/tcp/4001/ipfs/QmSoLju6m7xTh3DuokvT3886QRYqxAzb1kShaanJgW36yx
ipfs bootstrap rm /ip4/104.236.176.52/tcp/4001/ipfs/QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z
ipfs bootstrap rm /ip4/104.236.179.241/tcp/4001/ipfs/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM
ipfs bootstrap rm /ip4/104.236.76.40/tcp/4001/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64
ipfs bootstrap rm /ip4/128.199.219.111/tcp/4001/ipfs/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu
ipfs bootstrap rm /ip4/162.243.248.213/tcp/4001/ipfs/QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm
ipfs bootstrap rm /ip4/178.62.158.247/tcp/4001/ipfs/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd
ipfs bootstrap rm /ip4/178.62.61.185/tcp/4001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3
```

Then kill the daemon (ctrl+c). Then re-start it:
```
ipfs daemon
```

Then (while IPFS is not stuck constantly dialing the wrong port and backing off as in #2543), connect to one of the bootstrap peers on the alternate port:

```

**Comment by sterpe** [comment]
Perfect.  Thanks.

`ipfs bootstrap | sed 's/.*/ipfs bootstrap rm &/' | bash`

**Comment by jbenet** [comment]
Thanks for reporting, everyone. Cc @lgierth @whyrusleeping

**Comment by kbtombul** [comment]
I think I found the offending peers, all the others can be connected with ipfs swarm connect but these two, they either time-out or give connection refused error:
/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ
/ip4/104.236.176.52/tcp/4001/ipfs/QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z

Once I removed those two everything went back to normal.

**Comment by whyrusleeping** [comment]
what operating systems is everyone running with the delayed connection issue?

**Comment by sterpe** [comment]
El capitan, building from src, but same when using either 32 or 64 bit binaries

**Comment by mitzip** [comment]
pre-built 64bit linux

**Comment by whyrusleeping** [comment]
Thanks for reporting, found the issue. the old 0.4.0-dev nodes are dialing to the new 0.4.0 release and hanging (since they don't understand eachother). This causes further accepts to block.

I've pushed a patch to libp2p here: https://github.com/ipfs/go-libp2p/pull/35 and will merge that into ipfs once its reviewed (hopefully tomorrow)

**Comment by RichardLitt** [comment]
Is this fixed, now that that was merged?

**Comment by gavinmcdermott** [comment]
Also building from source today (OS X El Capitan as well); having the same issue with no peer response.
```
$ ipfs swarm peers
$
```

**Comment by whyrusleeping** [comment]
@gavinmcdermott what commit are you building from?

**Comment by gavinmcdermott** [comment]
@whyrusleeping thanks for the quick reply. I was building from the latest version. Interim update: uninstalled and reinstalled from the [distribution site](https://ipfs.io/ipns/dist.ipfs.io/#go-ipfs); peers connected now and able to retrieve. 

```
$ ipfs swarm peers
/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ
/ip4/178.62.61.185/tcp/4001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3
```

Will attempt to reproduce the issue at home tonight and let you know if the same thing arises.

**Comment by whyrusleeping** [comment]
I'm quite certain this issue has been fixed in the 0.4.1 release.

@gavinmcdermott if you continue to experience this after updating to 0.4.1, please feel free to open a new issue and reference this one in it.

**Comment by fbucci1** [comment]
$ ipfs swarm peers
-> No response
$ ipfs swarm connect /ip4/178.62.61.185/tcp/14001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3
-> Error: connect QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3 failure: dial attempt failed: context deadline exceeded

**Comment by Gcaufy** [comment]
Tried everything above, still do not work

```
➜  /Users/gcaufy
 $ ipfs swarm peers
➜  /Users/gcaufy
 $ lsof -i -n -P | grep ipfs
ipfs      60536 gcaufy   11u  IPv4 0xf116cfd5406528c3      0t0  UDP *:5353
ipfs      60536 gcaufy   14u  IPv4 0xf116cfd53df152bb      0t0  TCP *:4001 (LISTEN)
ipfs      60536 gcaufy   16u  IPv6 0xf116cfd569041573      0t0  TCP *:4001 (LISTEN)
ipfs      60536 gcaufy   18u  IPv4 0xf116cfd55aa41ffb      0t0  TCP 127.0.0.1:5001 (LISTEN)
ipfs      60536 gcaufy   19u  IPv4 0xf116cfd53e960c1b      0t0  TCP 127.0.0.1:8080 (LISTEN)
➜  /Users/gcaufy
 $ ipfs swarm connect /ip4/178.62.61.185/tcp/14001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3
Error: connect QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3 failure: dial attempt failed: context deadline exceeded
➜  /Users/gcaufy
 $ ipfs --version
ipfs version 0.4.13
```

**Comment by fbucci1** [comment]
What are the network requirements for IPFS? Do you need to have a public IP? Do you need to open any ports por allowing remote connections to it?

**Comment by GitDoggo** [comment]
I have a similar issue, `ipfs swarm peers` returns nothing. I tried waiting a few hours, with no better success.

I am suspecting my very restrictive corporate proxy is blocking it, how can I see if it's interfering or not? (I read [this](https://github.com/ipfs/ipfs/issues/209) and all the linked issues and nothing helped.)
